### ISSUE TITLE
[BUG] harvester.refresh_plots() should not be blocking?

### ISSUE THREAD

**Issue by mattsahr** [Chia-Network/chia-blockchain#5208]
I see long delays, getting "Looking up qualities" messages in the debug.log.  I read through the other issues about similar long lookups, and it seemed like the delays increase with the size of the farm.  Which is to say -- the reported delays are not actually about a specific hard drive lookup.  

In this file: `/chia/harvester/harvester_api.py`
every time there's a challenge, the harvester checks to see if it should refresh all the plots.

```
# LINE ~ 72
# Refresh plots to see if there are any new ones
if start - self.harvester.last_load_time > self.harvester.plot_load_frequency:
	await self.harvester.refresh_plots()
	self.harvester.last_load_time = time.time()
```

Is it necessary to use `await` for the `self.harvester.refresh_plots()` call?  Isn't that going to block everything subsequent in the loop?

I have over 1800 plots, and many of them are on NAS connections (I know, that's apparently not ideal) but the subequent error reports keep telling me that a whole batch of plot lookups took too long.  

```
# LINE ~ 196
# Concurrently executes all lookups on disk, to take advantage of multiple disk parallelism
total_proofs_found = 0
for filename_sublist_awaitable in asyncio.as_completed(awaitables):
	filename, sublist = await filename_sublist_awaitable
	time_taken = time.time() - start
	if time_taken > 5:
		self.harvester.log.warning(
			f"Looking up qualities on {filename} took: {time_taken}. This should be below 5 seconds "
			f"to minimize risk of losing rewards."
		)
```

So, it's good that the lookups are happening in parallel, but this takes place AFTER the blocking `self.harvester.refresh_plots()` call at the top of the process.  

I don't know what the solution is.  Can `refesh_plots()` run in its own thread, divorced from the lookup loop?  At minimum, the log files should say something like:

"Probably this lookup itself was just fine, but the system-wide collection of refesh_plots() pushed your response time up to XXX seconds."

**Comment by x1957** [comment]
Yes, the same problem , I have 9000 plots, when refresh_plots it tooks long time.
<code> refesh_plots()</code> should not block harverster.

**Comment by ctrlaltdel** [comment]
I've pushed some work in progress code to address this issue, testing welcome :)

**Comment by mattsahr** [comment]
Took me a while to get around to testing.  But I've been testing the changes in [[PR #5584](https://github.com/Chia-Network/chia-blockchain/pull/5584)] for a few days on a linux farmer, and my extremely long "looking up qualities" reports are eliminated.

I still get some  "looking up qualities" warnings in the 5 to 7 seconds range, but I think that's just because of the multiple consumer-level NAS boxes I have in the mix.

So for me, this fix WORKS.

**Comment by mattsahr** [comment]
This issue is still outstanding.  Every time I update, I am obliged to disable the harvester line ~75  in /chia/harvester/harvester_api.py  like so...

```
#  SKIP THIS  await self.harvester.refresh_plots()
```

**Comment by ctrlaltdel** [comment]
The patch in https://github.com/Chia-Network/chia-blockchain/pull/5584 has been working fine for the last two months, would love to have someone review it.

**Comment by emlowe** [comment]
Inspired by the work of @ctrlaltdel in #5584 - we made some changes in #7848 - we are trying to batch up the plot checks in refresh_plots (part of this is to support sending batch updates to the farmer and UI for 1000s of plots) - this also allows the processing of SPs to "intermingle" with the plot loading. Certain adventurous types could try out the new PR (caveat, it has had limited testing)
